lib/rucio/core/meta_conventions.py

```diff
@@ -32,9 +32,9 @@
 
 
 @transactional_session
-def add_key(key, key_type, value_type=None, value_regexp=None, *, session: "Session"):
+def add_key(key: str, key_type: Union[KeyType, str], value_type: Optional[str] = None, value_regexp: Optional[str] = None, *, session: "Session"):
     """
-    Adds a new allowed key.
+    Add an allowed key for DID metadata.
```

Comment:
I agree, that makes sense to me. It was difficult to figure out wording for some of the documentation even trying to differentiate a metadata key and a metadata conventions key.

Reply:
Something of note while I'm playing around with the tables - after trying to convert the queries in meta_conventions to sqlalchemy 2.0 (re #5247) I noticed a lot of queries that here break when you do this because of foreign key constraints. (e.g.: `(psycopg2.errors.ForeignKeyViolation) update or delete on table "did_keys" violates foreign key constraint "DID_MAP_KEYS_FK" on table "did_key_map"` is raised for any interaction that involves both of the tables, specifically for updating the value of an existing key). 

Is there an additional statements that need to be added to the sqla2.0 queries, is there something running amuck in the model definitions, or are the existing tests just doing unintended things?

Reply:
Yes, for example swapping 
```
session.query(models.DIDMetaConventionsConstraints.value).\
                filter_by(key=key).\
                filter_by(value=meta[key]).\
                one()

```

for 
```
            statement = select(
                models.DIDMetaConventionsConstraints.value
            ).where(
                and_(models.DIDMetaConventionsConstraints.value == meta[key], models.DIDMetaConventionsConstraints.key == key)
            )
            session.execute(statement).one()

```

Causes this, as does swapping out the `del_key` functionality: 
```
   session.query(models.DIDMetaConventionsKey).filter(key == key).delete()
```
-> 
```
     statement = delete(
             models.DIDMetaConventionsKey
         ).where(
             models.DIDMetaConventionsKey.key == key
         )
     session.execute(statement)
```

Reply:
Nevermind, figured it out. I hadn't been re-syncing the db tables between running tests so things got confused.